application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -0,0 +1,196 @@
+            .filter(channel -> modMailChannelNamePredicate.test(channel.getName()))
+            .findAny();
+        if (modMailAuditLog.isEmpty()) {
+            event
+                .reply("Sorry, there was an issue sending your report to the moderators. We are "
+                        + "investigating.")
+                .queue();
+            logger.warn(
+                    "Cannot find the designated modmail channel in server by id {} with the pattern {}",
+                    guildID, configModMailChannelPattern);
+        }
+        return modMailAuditLog;
+    }
+
+    private boolean handleIsOnCooldown(long userID, MessageContextInteractionEvent event) {
+        if (!isAuthorOnCooldown(userID)) {
+            return false;
+        }
+        event.reply("Can only be used once per %s minutes.".formatted(COOLDOWN_DURATION_VALUE))
+            .setEphemeral(true)
+            .queue();
+        return true;
+    }
+
+    private boolean isAuthorOnCooldown(long userId) {
+        return Optional.ofNullable(authorToLastReportInvocation.getIfPresent(userId))
+            .map(sinceCommandInvoked -> sinceCommandInvoked.plus(COOLDOWN_DURATION_VALUE,
+                    COOLDOWN_DURATION_UNIT))
+            .filter(Instant.now()::isBefore)
+            .isPresent();
+    }
+
+    private MessageCreateAction createModMessage(String modalMessage, long userID,
+            String reportedMessage, String reportedMessageUrl, TextChannel modMailAuditLog) {
+        MessageEmbed embed = new EmbedBuilder().setTitle("Report")
+            .setDescription(("""
+                    Reported Message: **%s**
+                    [Reported Message URL](%s)
+                    Reason: **%s**""").formatted(reportedMessage, reportedMessageUrl, modalMessage))
+            .setAuthor("Author ID: %d".formatted(userID))
+            .setColor(AMBIENT_COLOR)
+            .build();
+
+        return modMailAuditLog.sendMessageEmbeds(embed);
+    }
+
+    private void sendModMessage(ModalInteractionEvent event, List<String> args,
+            TextChannel modMailAuditLog) {
+        InteractionHook hook = event.getHook();
+
+        String reportedMessage = args.get(0);
+        String reportedMessageUrl = args.get(1);
+
+        event.deferReply().setEphemeral(true).queue();
+        String modalMessage = event.getValue(REPORT_REASON_INPUT_ID).getAsString();
+        long userID = event.getUser().getIdLong();
+        createModMessage(modalMessage, userID, reportedMessage, reportedMessageUrl, modMailAuditLog)
+            .mapToResult()
+            .map(this::createUserReply)
+            .flatMap(hook::editOriginal)
+            .queue();
+    }
+
+    private String createUserReply(Result<Message> result) {
+        if (result.isSuccess()) {
+            return "Thank you for reporting this message. A moderator will take care of the matter as soon as possible.";
+        }
+        logger.warn("Unable to forward a message report to modmail channel.");
```

Comment:
attach the failure reason to the log. `...modmail channel.", result.getFailure());`